### PR TITLE
Add support for external links.

### PR DESCRIPTION
**Pull request by leebickmtu**
Add support for external links. Let browser open in new tab/window based on it's preference by using window.open().

@evanjd, one of the mentioned features

### CODE REVIEW

**Comment by leebickmtu** [comment]
This is now ready for review.

File: px-app-nav.es6.js [hunk 1]
```diff
@@ -56,14 +56,21 @@
        * and `label` defined. If an `icon` or `children` property is defined
        * on a subitem, those properties will be ignored.
        *
+       * The following optional properties can be used for any-level items:
+       *
+       * - {Boolean} isExternal - A boolean value for if this item should trigger
```
Comment (evanjd) [inline px-app-nav.es6.js]: `isExternal` only makes sense if a `link` prop is set (and `link` does not work if `isExternal` is missing). This makes this prop redundant.

I think we should remove this prop and replace it with:

- `href`: a string that contains whatever URL this item should navigate to
- `window`: an optional prop that determines what window opens the link. If it's unset, navigation should occur on the current page (no new tab/window etc). If it's set, that value is passed in as the 2nd param to `window.open`.

Thoughts?

File: px-app-nav.es6.js [hunk 2]
```diff
@@ -56,14 +56,21 @@
        * and `label` defined. If an `icon` or `children` property is defined
        * on a subitem, those properties will be ignored.
        *
+       * The following optional properties can be used for any-level items:
```
Comment (evanjd) [inline px-app-nav.es6.js]: > ...can be used for any-level items

It only works on items with no children.

File: px-app-nav.es6.js (hunk 1 above)
Comment (leebickmtu) [reply px-app-nav.es6.js]: I like that idea, but disagree about the unset window behavior. I think most use uses of an external link would expect a new tab to open.

I would say use the identical behavior the `window.open()` spec outlines 
https://developer.mozilla.org/en-US/docs/Web/API/Window/open

Which is that empty `windowName` defaults to new tab/window.
`windowName='_self'` navigates the current window.
And `windowName='whatever'` open a new tab/window on first click and then renavigates that named window while it's still open.

File: px-app-nav.es6.js (hunk 2 above)
Comment (leebickmtu) [reply px-app-nav.es6.js]: Good catch, will update the comment

File: px-app-nav.es6.js (hunk 1 above)
Comment (evanjd) [reply px-app-nav.es6.js]: Yep, I agree.

Comment (leebickmtu) [reply px-app-nav.es6.js]: Changes we discussed are committed as well as the comment fix. Tried them on my local project, as worked well.

**Comment by leebickmtu** [comment]
@evanjd Change looks good to me.

**Comment by evanjd** [comment]
@leebickmtu 2.3.0 released with this change: https://github.com/predixdesignsystem/px-app-nav/releases/tag/v2.3.0

**Pull request merged**